### PR TITLE
update cdp genesis state and migrations

### PR DESCRIPTION
**Pull request by karzak**
* Updates `GenesisState` for CDP module to reflect accumulator based interest.
* Adds migrations for CDPs from kava-4 (v0.12) to v0.13+

### CODE REVIEW

**Review by rhuairahrighairidh — CHANGES_REQUESTED** [review]
In `cdp/genesis.go` ln 84, the total principle is calculated by summing the cdps' `TotalPrinciple()` but this won't include the interest accumulated since the cdp was last modified. So in moving from kava-5 to kava-6 some TotalPrincipal will be dropped.

I think either the raw TotalPrincipal amount needs to be stored in genesis, or `SynchronizeInterest` needs to be run on all cdps on export.

File: x/cdp/keeper/deposit.go [hunk 1]
```diff
@@ -21,6 +21,14 @@ func (k Keeper) DepositCollateral(ctx sdk.Context, owner, depositor sdk.AccAddre
...
 		return sdkerrors.Wrapf(types.ErrCdpNotFound, "owner %s, collateral %s", owner, collateralType)
 	}
+	err = k.ValidateBalance(ctx, collateral, depositor)
+	if err != nil {
+		return err
+	}
+
+	oldCollateralToDebtRatio := k.CalculateCollateralToDebtRatio(ctx, cdp.Collateral, cdp.Type, cdp.GetTotalPrincipal())
+
+	cdp = k.SynchronizeInterest(ctx, cdp)
```
[the remark this answers is not in the document]
Comment (karzak) [reply x/cdp/keeper/deposit.go]: Resolved in 25d09b7

**Comment by karzak** [comment]
> In `cdp/genesis.go` ln 84, the total principle is calculated by summing the cdps' `TotalPrinciple()` but this won't include the interest accumulated since the cdp was last modified. So in moving from kava-5 to kava-6 some TotalPrincipal will be dropped.
> 
> I think either the raw TotalPrincipal amount needs to be stored in genesis, or `SynchronizeInterest` needs to be run on all cdps on export.

Good catch. Updated `GenesisState` to include a total principal argument, as it's no longer the case that CDP principal and total principal add up exactly.

**Review by rhuairahrighairidh — APPROVED** [review]
nice, looks good to me

**Pull request merged**